packages/@office-iss/react-native-win32/src/Libraries/Utilities/Dimensions.win32.js

```diff
@@ -6,29 +6,126 @@
  * @flow
  */
 
-import {type EventSubscription} from '../vendor/emitter/EventEmitter';
+import EventEmitter, {
+  type EventSubscription,
+} from '../vendor/emitter/EventEmitter';
+import RCTDeviceEventEmitter from '../EventEmitter/RCTDeviceEventEmitter';
+import NativeDeviceInfo, {
+  type DisplayMetrics,
+  type DisplayMetricsAndroid,
+  type DimensionsPayload,
+} from './NativeDeviceInfo';
+import invariant from 'invariant';
 
+const eventEmitter = new EventEmitter<{
+  change: [DimensionsPayload],
+}>();
+let dimensionsInitialized = false;
+let dimensions: DimensionsPayload;
+
+/**
+ * While a global Dimensions object for window and screen dimensions is too simple for Win32,
+ * attached to this object is also fontScale which is a system global value.  We expose this value
+ * for large text scaling support while zeroing out other window dimension information.  These zeroes will
+ * cause rendering issues if used but should avoid runtime failures in JS.
+ */
 class Dimensions {
-  static get(dim: string): Object {
-    throw new Error(
-      'Having a global Dimensions object is too simplistic for Win32, so this API does not work',
-    );
+  /**
+   * NOTE: `useWindowDimensions` is the preferred API for React components.
+   *
+   * Initial dimensions are set before `runApplication` is called so they should
+   * be available before any other require's are run, but may be updated later.
+   *
+   * Note: Although dimensions are available immediately, they may change (e.g
+   * due to device rotation) so any rendering logic or styles that depend on
+   * these constants should try to call this function on every render, rather
+   * than caching the value (for example, using inline styles rather than
+   * setting a value in a `StyleSheet`).
+   *
+   * Example: `const {height, width} = Dimensions.get('window');`
+   *
+   * @param {string} dim Name of dimension as defined when calling `set`.
+   * @returns {DisplayMetrics? | DisplayMetricsAndroid?} Value for the dimension.
+   */
+  static get(dim: string): DisplayMetrics | DisplayMetricsAndroid {
+    invariant(dimensions[dim], 'No dimension set for key ' + dim);
+    return dimensions[dim];
   }
 
-  static set(dims: $ReadOnly<{[key: string]: any, ...}>): void {
-    throw new Error(
-      'Having a global Dimensions object is too simplistic for Win32, so this API does not work',
-    );
+  /**
+   * This should only be called from native code by sending the
+   * didUpdateDimensions event.
+   *
+   * @param {DimensionsPayload} dims Simple string-keyed object of dimensions to set
+   */
+  static set(dims: $ReadOnly<DimensionsPayload>): void {
+    let {screen, window} = dims;
+    const {windowPhysicalPixels} = dims;
+    if (windowPhysicalPixels) {
+      window = {
+        width: 0,
```

Comment:
Can we maybe throw an error here when trying to use anything but fontScale? Otherwise, it wouldn't be clear why it's not working.

Reply:
+1.  I really like the idea of keeping these APIs throwing to avoid any uses sneaking in.

Reply:
Thanks for the recommendation. I changed the native implementation to keep the width, height, and scale values undefined and used `invariant` in the `get` function to throw an error if those values are retrieved. Is that sufficient?

Reply:
On second thought, I don't think that works well. The `get` function retrieves the dimensions for either 'window' or 'screen' and not a specific display metric (like 'width', 'height', or 'scale') so an error would be thrown each time someone uses `get` or `useWindowsDimensions` even if they're not using the unimplemented values.

I'm not sure how to throw an error when 'width', 'height', or 'scale' is referenced. Does anyone have a suggestion? I could use a warning instead but that wouldn't prevent someone from using the APIs.

Reply:
rather than 0s we only define fontScale in each. which will cause a runtime error

Reply:
Okay, I redefined the unimplemented values from 0s to undefined values so now a render error is thrown when someone tries to use them.